### ISSUE TITLE
Remote containers over SSH not working

### ISSUE THREAD

**Issue by adns44** [microsoft/vscode#194879]
Does this issue occur when all extensions are disabled?: Not understable

- VS Code Version: 1.83.0 (system setup)
- OS Version: Windows_NT x64 10.0.22621

Steps to Reproduce:

1. Open VsCode and connect to a remote SSH host
2. Try to open a container (if exists the neccessary files) in a folder or try to attach a running container via command palette
3. You will see the following error message:
"Command 'Dev Containers: Attach to Running Container...' resulted in an error
The operation was aborted
OK "

**Comment by Rairot** [comment]
I'm experiencing the same issue, it started appearing after i removed my last dev container and tried creating a new one today.

```
Version: 1.83.0
Commit: e7e037083ff4455cf320e344325dacb480062c3c
Date: 2023-10-03T16:11:49.843Z
Electron: 25.8.4
ElectronBuildId: 24154031
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Linux x64 6.5.4-76060504-generic
```
The client is using the pre-release version of remote - SSH, the Release version produces an extra error described here https://github.com/microsoft/vscode/issues/192658
*  Remote - SSH - v0.107.2023092715 
* Dev Containers - v0.316.0 or  v0.315.0  (neither work)


Extension Host Output:
```
2023-10-05 13:08:40.934 [error] Error: The operation was aborted
	at Socket.<anonymous> (/home/me/.vscode/extensions/ms-vscode-remote.remote-ssh-0.107.2023092715/out/extension.js:2:137710)
	at Object.onceWrapper (node:events:628:26)
	at Socket.emit (node:events:525:35)
	at emitErrorNT (node:internal/streams/destroy:151:8)
	at emitErrorCloseNT (node:internal/streams/destroy:116:3)
	at process.processTicksAndRejections (node:internal/process/task_queues:82:21) remote-containers.rebuildNoCacheAndReopenInContainer {"value":"ms-vscode-remote.remote-containers","_lower":"ms-vscode-remote.remote-containers"}
```

I removed the host side `$HOME/.vscode-server ` folders, killed any lingering processes and rebooted the server but it did not help.
Using the same extensions and creating the devcontainer on my local machine works just fine.

Steps to reproduce:
1) Connect to remote host over ssh
2) Command `Dev Containers: Reopen in Container`

**Comment by rivershah** [comment]
Same here, this is really poor unit testing during release process. Please fix as collectively this is a huge amount of developer hours wasted.

**Comment by UrkoAT** [comment]
Having the same problem.
`Version: 1.83.0
Commit: e7e037083ff4455cf320e344325dacb480062c3c
Date: 2023-10-03T16:11:49.843Z
Electron: 25.8.4
ElectronBuildId: 24154031
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Linux x64 6.2.0-34-generic`

Rebooting, clearing .vscode-server and .vscode-remote-containers, uninstalling and reinstalling the extensions doesnt work either. Neither switching to pre release. All the dev container commands wont work. In fact, they disconnect Remote - SSH and it reconects.

**Comment by kristireispass** [comment]
Same here.

`
Version: 1.83.0 (user setup)
Commit: e7e037083ff4455cf320e344325dacb480062c3c
Date: 2023-10-03T16:12:16.321Z
Electron: 25.8.4
ElectronBuildId: 24154031
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.19045
`

**Comment by GillesBallegeerVintecc** [comment]
Same here. Is there a way to revert the remote server version?

**Comment by norazamaziz** [comment]
Same here

**Comment by JNaranjo-Alcazar** [comment]
Same here

**Comment by norazamaziz** [comment]
> Same here

fall back remote-ssh to previous version solve my problem

**Comment by chrmarti** [comment]
Continuing in https://github.com/microsoft/vscode-remote-release/issues/9064. Thanks.

**Comment by daniel-trnk** [comment]
> > Same here
> 
> fall back remote-ssh to previous version solve my problem

Which version? I have tried rolling back but is still has the same issue:
```
$ code --version
1.83.0
e7e037083ff4455cf320e344325dacb480062c3c
x64

$ code --list-extensions --show-versions | grep remote
ms-vscode-remote.remote-containers@0.314.0
ms-vscode-remote.remote-ssh@0.106.4
ms-vscode-remote.remote-ssh-edit@0.86.0
ms-vscode.remote-explorer@0.4.1
```

**Comment by apivovarov** [comment]
Same issue. 

Mac - Connect to host (Linux) - attach to Running container

Extension Host Logs on Linux show
```
2023-10-05 20:24:05.851 [error] Error: The operation was aborted
	at Socket.<anonymous> (/Users/aaa/.vscode/extensions/ms-vscode-remote.remote-ssh-0.106.4/out/extension.js:2:137555)
	at Object.onceWrapper (node:events:628:26)
	at Socket.emit (node:events:525:35)
	at emitErrorNT (node:internal/streams/destroy:151:8)
	at emitErrorCloseNT (node:internal/streams/destroy:116:3)
	at process.processTicksAndRejections (node:internal/process/task_queues:82:21) remote-containers.attachToRunningContainer {"value":"ms-vscode-remote.remote-containers","_lower":"ms-vscode-remote.remote-containers"}
2023-10-05 20:24:06.854 [error] TypeError: Cannot use 'in' operator to search for 'result' in 72
	at /Users/aaa/.vscode/extensions/ms-vscode-remote.remote-ssh-0.106.4/out/extension.js:2:137281
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**Comment by Melkaz** [comment]
The above workaround works, thank you <3

**Comment by hamsjelv** [comment]
Theres is just released a new update 1.83.1, which fixes the problem.

if you used this workaround:
```
Solution/Workaround:
Enable Remote.SSH: Use Exec Server in Settings
```

Disable the setting again, for smooth sailing.
